File: lib/validation/turno.schema.ts
import { z } from 'zod';

export const turnoSchema = z.object({
  paciente_id: z.coerce.number().int().positive(),
  medico_id: z.coerce.number().int().positive(),
  fecha_hora: z.string().min(10), // ISO
  duracion_min: z.coerce.number().int().positive().default(30),
  motivo: z.string().optional().or(z.literal('')),
  observaciones: z.string().optional().or(z.literal('')),
});

export type TurnoInput = z.infer<typeof turnoSchema>;

File: backend/app/schemas/turno_schema.py
from pydantic import BaseModel, ConfigDict


class TurnoCreate(BaseModel):
    paciente_id: int | str | None = None
    medico_id: int | str | None = None
    fecha_hora: str | None = None
    duracion_min: int | str | None = None
    estado_turno_id: int | str | None = None
    motivo: str | None = ""
    observaciones: str | None = ""

    model_config = ConfigDict(extra="allow")


class TurnoUpdate(BaseModel):
    estado_turno_id: int | str | None = None
    fecha_hora: str | None = None

    model_config = ConfigDict(extra="allow")
